src/process/message_processor.go

```diff
@@ -5,70 +5,97 @@ import (
 	"BGPAlert/common"
 	"BGPAlert/config"
 	"fmt"
+	"strconv"
 	"time"
 )
 
 // Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
-func ProcessBGPMessages(msgChannel chan common.BGPMessage, config *config.Configuration) error {
-	maximumBuckets := config.WindowSize
+// Constantly read messages from channel, build up windows with frequency maps, calling analysis on windows when full
+func ProcessBGPMessages(msgChannel chan common.Message, config *config.Configuration) error {
```

Comment:
Does it make sense to just pass the window size value rather than the entire config?

Reply:
As a follow up - probs makes the most sense to establish the default value somewhere in the config validation method as opposed to here.